### PR TITLE
lib id_prefix: look for divergent changes outside short prefix set 

### PR DESCRIPTION
**Pull request by ilyagr**
Fixes https://github.com/martinvonz/jj/issues/2476.

Previously, if there was a change id match within the short prefix
lookup set, `jj` would not look for commits with that same change id
outside the short prefix set. So, it wouldn't find the conflicted
commits for a commit with a divergent (AKA conflicted) change id.

# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- n/a I have updated the documentation (README.md, docs/, demos/)
- n/a I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: lib/tests/test_id_prefix.rs [hunk 1]
```diff
@@ -253,3 +253,149 @@ fn test_id_prefix() {
...
+                .set_description(description)
+                .set_author(signature.clone())
+                .set_committer(signature);
+            if let Some(change_id) = change_id {
+                builder = builder.set_change_id(change_id);
+            }
+            builder.write().unwrap()
+        };
+
+    let first_commit = create_commit_with_change_id(root_commit_id, "first", None);
```
Comment (yuja) [inline lib/tests/test_id_prefix.rs:291]: nit: maybe better to hardcode the change_id instead of relying on the seed?

**Review by yuja — APPROVED** [review]
Thanks.

File: lib/tests/test_id_prefix.rs (hunk 1 above)
Comment (ilyagr) [reply lib/tests/test_id_prefix.rs:291]: Good point, this simplifies a few things.

**Comment by ilyagr** [comment]
Thank you, Yuya!

**Pull request merged**